WcaOnRails/app/controllers/admin_controller.rb

```diff
@@ -75,4 +75,16 @@ def person_data
       dob: @person.dob,
     }
   end
+
+  def compute_auxiliary_data
+    @auxiliary_data_start = Timestamp.find_or_create_by(name: 'auxiliary_data_start')
+    @auxiliary_data_end = Timestamp.find_or_create_by(name: 'auxiliary_data_end')
+  end
+
+  def do_compute_auxiliary_data
+    # Reset the end date. The start date is updated directly before the computation.
+    Timestamp.find_or_create_by(name: 'auxiliary_data_end').update date: nil
+    ComputeAuxiliaryData.perform_later
```

Comment:
Ah yes, I thought about deferring, but forgot to mention it. One idea using exceptions:

```ruby
class TimedApplicationJob
  class DeferJob
      def initialize(wait....)
        # i think you get the idea ...
      end
  end

  def self.defer_job_until(...)
    raise DeferJob.new(wait: 10.minutes)
  end

  around_perform do |job, block|
    start_timestamp.touch :date
    end_timestamp.update! date: nil
    try
      block.call
      end_timestamp.touch :date
    rescue TimedApplicationJob::DeferJob => e
      job.wait(e.wait)
    end
  end
end
```